### ISSUE TITLE
cannot build on armbian ubuntu kernal 5.10.110

### ISSUE THREAD

**Issue by EricYin** [nicman23/dkms-hid-nintendo#64]
I have orangepi5 board with armbian linux installed (Armbian 23.02 Jammy Gnome, kernel 5.10.110). but when i run "sudo dkms build nintendo -v 3.2" on this orangepi 5, i got error below. is anyone know how i can fix it? I installed linux-headers-legacy-rockchip-rk3588_23.02.2_arm64.deb to build this driver before i got this error. it looks the make behavir is changed after kernal 5.10, as someone mentioned that module.lds is missing and only modules.lds.S in header scripts folder. how can i get module.lds without building the whole kernel by myself?
https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1906131

DKMS make.log for nintendo-3.2 for kernel 5.10.110-rockchip-rk3588 (aarch64)
Sat Apr  8 01:47:55 PM CST 2023
make: Entering directory '/usr/src/linux-headers-5.10.110-rockchip-rk3588'
  CC [M]  /var/lib/dkms/nintendo/3.2/build/src/hid-nintendo.o
  MODPOST /var/lib/dkms/nintendo/3.2/build/src/Module.symvers
make[2]: *** No rule to make target 'scripts/module.lds', needed by '/var/lib/dkms/nintendo/3.2/build/src/hid-nintendo.ko'.  Stop.
make[2]: *** Waiting for unfinished jobs....
  CC [M]  /var/lib/dkms/nintendo/3.2/build/src/hid-nintendo.mod.o
make[1]: *** [scripts/Makefile.modpost:175: __modpost] Error 2
make: *** [Makefile:1822: modules] Error 2
make: Leaving directory '/usr/src/linux-headers-5.10.110-rockchip-rk3588'

**Comment by nicman23** [comment]
i would suggest just going with a backport kernel as upstream includes this driver. also i do not have an arm system to test anything on.

**Comment by EricYin** [comment]
> i would suggest just going with a backport kernel as upstream includes this driver. also i do not have an arm system to test anything on.


Thanks, I figured out the module.lds missing is a bug of armbian build code. I switch to use its master build code and get module.lds. then I can build and install nintendo hid driver. and also installed joycond. my switch pro controller can be paired and connect in my ubuntu arm OS, that is great. one problem I found is each time my ubuntu os is restarted, i cannot connect to pro controller directly but windows 10 can connect swith pro controller automatically when pro controller wake up. to connect in ubuntu I have to make switch pro controller into pair mode again and connect then. any thing I missed?

**Comment by EricYin** [comment]
some update here: I bought new bluetooth adapter (real csr8510 chip). it is better than old one. but it seem after several minunites( playing the game), connection is dropped. the log of bluetooth said: host is down, cannot read hidinfo, something like this. the switch pro controller works great in windows 10. so after several days effort, I have to say hard to play with arm linux using swith pro controller.

**Comment by nfp0** [comment]
> connection is dropped

@EricYin It's a known issue that it's trying to be solved. It happens during rumble.
You might want to subscribe to this issue from the kernel: https://github.com/DanielOgorchock/linux/issues/33

**Comment by EricYin** [comment]
> > connection is dropped
> 
> @EricYin It's a known issue that it's trying to be solved. It happens during rumble. You might want to subscribe to this issue from the kernel: [DanielOgorchock/linux#33](https://github.com/DanielOgorchock/linux/issues/33)

Thanks, rumble is impartant for game, hope this can be fixed.

**Comment by nicman23** [comment]
one workaround is to play with the controller plugged in

**Comment by nfp0** [comment]
Another possible workaround is to use the controller through Steam, as the Steam driver does not seem to cause disconnects.

**Comment by nicman23** [comment]
does armbian have steam? wut

**Comment by EricYin** [comment]
> does armbian have steam? wut

no, I did not see steam on armbian(box64/box86 may be a workaround, but I didnot try). I tried moonlight client streaming steam game from remote machine. lose bluetooth connection after several minutes..if i disable rumble, it seems i can play a long time.

**Comment by nfp0** [comment]
> does armbian have steam? wut

My bad, I forgot he was talking specifically about armbian.
